jplag/src/main/java/de/jplag/JPlag.java

```diff
@@ -100,24 +106,7 @@ private ComparisonStrategy initializeComparisonStrategy(final ComparisonMode com
         };
     }
 
-    private Language initializeLanguage() {
-        LanguageOption languageOption = this.options.getLanguageOption();
-
-        try {
-            Constructor<?> constructor = Class.forName(languageOption.getClassPath()).getConstructor(ErrorConsumer.class);
-            Object[] constructorParams = {errorCollector};
-
-            Language language = (Language) constructor.newInstance(constructorParams);
-
-            this.options.setLanguage(language);
-            this.options.setLanguageDefaults(language);
-            System.out.println("Initialized language " + language.getName());
-            return language;
-        } catch (NoSuchMethodException | SecurityException | ClassNotFoundException | InstantiationException | IllegalAccessException
-                | IllegalArgumentException | InvocationTargetException e) {
-            e.printStackTrace();
-            throw new IllegalStateException("Language instantiation failed:" + e.getMessage());
-        }
-
+    private Language loadLanguage(final ErrorCollector errorCollector, final String classPath) {
+        return Languages.provider(classPath).create(errorCollector);
```

Comment:
This became so short, there might not be a need for a dedicated method.

Reply:
This is the factory, I know this is short, but the goal here is to document the line, and hide construction details.

I will try to find another way to do that, like integrate it directly into `Languages`